diff --git a/scripts/mono-find-provides.in b/scripts/mono-find-provides.in
index c921006..155690d 100644
--- a/scripts/mono-find-provides.in
+++ b/scripts/mono-find-provides.in
@@ -33,7 +33,7 @@ fi
 # set LD_LIBRARY_PATH to ensure that libmono is found
 export LD_LIBRARY_PATH=$libdir${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 # and set MONO_PATH to ensure that mscorlib.dll can be found
-export MONO_PATH=$prefix/lib/mono/2.0
+export MONO_PATH=$prefix/lib/mono/4.5
 
 for i in "${monolist[@]}"; do
         ($bindir/monodis --assembly $i | awk '
diff --git a/scripts/mono-find-requires.in b/scripts/mono-find-requires.in
index f0ad60b..d7b6a32 100644
--- a/scripts/mono-find-requires.in
+++ b/scripts/mono-find-requires.in
@@ -50,7 +50,7 @@ esac
 # set LD_LIBRARY_PATH to ensure that libmono is found
 export LD_LIBRARY_PATH=$libdir${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 # and set MONO_PATH to ensure that mscorlib.dll can be found
-export MONO_PATH=$prefix/lib/mono/2.0
+export MONO_PATH=$prefix/lib/mono/4.5
 
 REQUIRES=$(
 	for i in "${monolist[@]}"; do
