Index: xamarin-pkg-mono/mono/mini/tramp-amd64.c
===================================================================
--- xamarin-pkg-mono.orig/mono/mini/tramp-amd64.c	2015-04-28 00:23:39.342850339 +0100
+++ xamarin-pkg-mono/mono/mini/tramp-amd64.c	2015-04-28 00:23:39.338850339 +0100
@@ -995,7 +995,7 @@
 	status_offset = MONO_THREADS_SYNC_MEMBER_OFFSET (status_offset);
 	nest_offset = MONO_THREADS_SYNC_MEMBER_OFFSET (nest_offset);
 
-	tramp_size = 96;
+	tramp_size = 128;
 
 	code = buf = mono_global_codeman_reserve (tramp_size);
 
