diff --git a/mono/mini/aot-compiler.c b/mono/mini/aot-compiler.c
index d75d3f934f2..885877869da 100644
--- a/mono/mini/aot-compiler.c
+++ b/mono/mini/aot-compiler.c
@@ -13279,7 +13279,7 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options,
 #endif
 	acfg->aot_opts.nunbox_arbitrary_trampolines = 256;
 	if (!acfg->aot_opts.llvm_path)
-		acfg->aot_opts.llvm_path = g_strdup ("");
+		acfg->aot_opts.llvm_path = g_strdup ("/usr/lib/mono/llvm/bin/");
 	acfg->aot_opts.temp_path = g_strdup ("");
 #ifdef MONOTOUCH
 	acfg->aot_opts.use_trampolines_page = TRUE;
