diff --git a/mono/mini/Makefile.am.in b/mono/mini/Makefile.am.in
index 60055ce9ac9..b91c74ec3a8 100755
--- a/mono/mini/Makefile.am.in
+++ b/mono/mini/Makefile.am.in
@@ -776,7 +776,7 @@ MemoryIntrinsics.dll: MemoryIntrinsics.il
 
 GENMDESC_OPTS=
 
-GENMDESC_PRG=python $(srcdir)/genmdesc.py $(target_define) $(srcdir)
+GENMDESC_PRG=python3 $(srcdir)/genmdesc.py $(target_define) $(srcdir)
 
 cpu-wasm.h: mini-ops.h cpu-wasm.md $(srcdir)/genmdesc.py
 	$(GENMDESC_PRG) cpu-wasm.h wasm_desc $(srcdir)/cpu-wasm.md
diff --git a/scripts/mono-heapviz b/scripts/mono-heapviz
index 27b6f7f6023..c4a4847bbe6 100755
--- a/scripts/mono-heapviz
+++ b/scripts/mono-heapviz
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Generate a heap visualization for SGen from the heap dump written by
 # mono if the MONO_GC_DEBUG is set to something like
