Index: xamarin-pkg-mono/mono/mini/mini-arm-tls.h
===================================================================
--- xamarin-pkg-mono.orig/mono/mini/mini-arm-tls.h
+++ xamarin-pkg-mono/mono/mini/mini-arm-tls.h
@@ -66,26 +66,6 @@ known_kernel_helper (void)
 #endif
 }
 
-static void
-dump_code (guint32 *ptr)
-{
-	char current_impl [256];
-	char hex [16];
-	int i;
-	guint32 page_mask = ~((guint32)mono_pagesize () - 1);
-
-	current_impl [0] = 0;
-	for (i = 0; i < 16; i++) {
-		/* Don't risk page fault since we don't know where the code ends */
-		if (((guint32)&ptr [i] & page_mask) != ((guint32)ptr & page_mask))
-			break;
-		sprintf (hex, "0x%x ", ptr [i]);
-		strcat (current_impl, hex);
-	}
-
-	g_warning (current_impl);
-}
-
 static MonoTlsImplementation
 mono_arm_get_tls_implementation (void)
 {
@@ -110,9 +90,6 @@ mono_arm_get_tls_implementation (void)
 		}
 	}
 
-	g_warning ("No fast tls on device. Using fallbacks. Current implementation : ");
-	dump_code (check_addr);
-
 	return (MonoTlsImplementation) { NULL, 0, FALSE, mono_fallback_get_tls_key, NULL, mono_fallback_set_tls_key, NULL };
 #endif
 }
Index: xamarin-pkg-mono/mono/mini/mini-arm.c
===================================================================
--- xamarin-pkg-mono.orig/mono/mini/mini-arm.c
+++ xamarin-pkg-mono/mono/mini/mini-arm.c
@@ -31,7 +31,7 @@
 #include "mini-gc.h"
 #include "mono/arch/arm/arm-vfp-codegen.h"
 
-#if (defined(HAVE_KW_THREAD) && defined(__linux__) && defined(__ARM_EABI__)) \
+#if (defined(HAVE_KW_THREAD) && defined(__linux__) && defined(__ARM_EABI__) && !defined(__PIC__)) \
 	|| defined(TARGET_ANDROID) \
 	|| (defined(TARGET_IOS) && !defined(TARGET_WATCHOS))
 #define HAVE_FAST_TLS
