diff --git a/mcs/class/Mono.Security/Mono.Security.X509/X509Store.cs b/mcs/class/Mono.Security/Mono.Security.X509/X509Store.cs
index 536782f..1821075 100644
--- a/mcs/class/Mono.Security/Mono.Security.X509/X509Store.cs
+++ b/mcs/class/Mono.Security/Mono.Security.X509/X509Store.cs
@@ -163,7 +163,7 @@ public void Import (X509Certificate certificate)
 			cspParams.KeyContainerName = CryptoConvert.ToHex (certificate.Hash);
 
 			// Right now this seems to be the best way to know if we should use LM store.. ;)
-			if (_storePath.StartsWith (X509StoreManager.LocalMachinePath))
+			if (_storePath.StartsWith (X509StoreManager.LocalMachinePath) || _storePath.StartsWith(X509StoreManager.NewLocalMachinePath))
 				cspParams.Flags = CspProviderFlags.UseMachineKeyStore;
 
 			ImportPrivateKey (certificate, cspParams);
@@ -338,7 +338,7 @@ private X509Certificate LoadCertificate (string filename)
 			// If privateKey it's available, load it too..
 			CspParameters cspParams = new CspParameters ();
 			cspParams.KeyContainerName = CryptoConvert.ToHex (cert.Hash);
-			if (_storePath.StartsWith (X509StoreManager.LocalMachinePath))
+			if (_storePath.StartsWith (X509StoreManager.LocalMachinePath) || _storePath.StartsWith(X509StoreManager.NewLocalMachinePath))
 				cspParams.Flags = CspProviderFlags.UseMachineKeyStore;
 			KeyPairPersistence kpp = new KeyPairPersistence (cspParams);
 
